update release notes (#723)
CEED/libCEED: doc/sphinx/source/releasenotes.rst

^^^^^^^^
* :ref:`example-petsc-elasticity` example updated with traction boundary conditions and improved Dirichlet boundary conditions.
* :ref:`example-petsc-elasticity` example updated with Neo-Hookean hyperelasticity in current configuration as well as improved Neo-Hookean hyperelasticity exploring storage vs computation tradeoffs.

> * :ref:`example-petsc-navier-stokes` example updated with Traveling Vortex test case which is the analytical solution to the Euler equations.

Reviewer: ```suggestion
* :ref:`example-petsc-navier-stokes` example updated with isentropic traveling vortex test case, an analytical solution to the Euler equations that is useful for testing boundary conditions, discretization stability, and order of accuracy.
```

> * :ref:`example-petsc-navier-stokes` example updated with a support for performing convergence study and plotting the obtained results.

Reviewer: ```suggestion
* :ref:`example-petsc-navier-stokes` example updated with support for performing convergence study and plotting order of convergence by polynomial degree.
```
